scicamera/controls.py

```diff
@@ -1,63 +1,45 @@
-#!/usr/bin/python3
 from __future__ import annotations
 
-import threading
 from typing import Set
 
 from libcamera import ControlType, Rectangle, Size
 
 
-def _framerates_to_durations_(framerates):
+def _framerates_to_durations(framerates):
     if not isinstance(framerates, (tuple, list)):
         framerates = (framerates, framerates)
     return (int(1000000 / framerates[1]), int(1000000 / framerates[0]))
 
 
-def _durations_to_framerates_(durations):
+def _durations_to_framerates(durations):
     if durations[0] == durations[1]:
         return 1000000 / durations[0]
     return (1000000 / durations[1], 1000000 / durations[0])
```

Comment:
Note that these are not inverses of eachother. Also, type hints would make it clear that one of them returns a `Tuple[int, int]`, while the other returns `Tuple[float, float]`.

Personally, I'd also limit the ability for users to supply variable types to avoid the `isinstance` checks, and instead rely on the user to look to type hints for the correct way to call the function.